spec/config-spec.ts

```diff
@@ -1,9 +1,168 @@
+      let called = false;
+      const results: any[] = [];
+
+      config.onUnhandledError = err => {
+        called = true;
+        expect(err).to.equal('bad');
+        done()
+      };
+
+      const source = new Observable<number>(subscriber => {
+        subscriber.next(1);
+        subscriber.error('bad');
+      });
+
+      source.subscribe(value => results.push(value));
+      expect(called).to.be.false;
+      expect(results).to.deep.equal([1]);
+    });
+
+    it('should call asynchronously if an error is emitted and not handled by the consumer in the empty case', (done) => {
+      let called = false;
+      config.onUnhandledError = err => {
+        called = true;
+        expect(err).to.equal('bad');
+        done()
+      };
+
+      const source = new Observable(subscriber => {
+        subscriber.error('bad');
+      });
+
+      source.subscribe();
+      expect(called).to.be.false;
+    });
+
+    it('should call asynchronously if a subscription setup errors after the subscription is closed by an error', (done) => {
+      let called = false;
+      config.onUnhandledError = err => {
+        called = true;
+        expect(err).to.equal('bad');
+        done()
+      };
+
+      const source = new Observable(subscriber => {
+        subscriber.error('handled');
+        throw 'bad';
+      });
+
+      let syncSentError: any;
+      source.subscribe({
+        error: err => {
+          syncSentError = err;
+        }
+      });
+
+      expect(syncSentError).to.equal('handled');
+      expect(called).to.be.false;
+    });
+
+    it('should call asynchronously if a subscription setup errors after the subscription is closed by a completion', (done) => {
+      let called = false;
+      let completed = false;
+
+      config.onUnhandledError = err => {
+        called = true;
+        expect(err).to.equal('bad');
+        done()
+      };
+
+      const source = new Observable(subscriber => {
+        subscriber.complete();
+        throw 'bad';
+      });
+
+      source.subscribe({
+        error: () => {
+          throw 'should not be called';
+        },
+        complete: () => {
+          completed = true;
+        }
+      });
+
+      expect(completed).to.be.true;
+      expect(called).to.be.false;
+    });
+
+    /**
+     * Thie test is added so people know this behavior is _intentional_. It's part of the contract of observables
+     * and, while I'm not sure I like it, it might start surfacing untold numbers of errors, and break
+     * node applications if we suddenly changed this to start throwing errors on other jobs for instances
+     * where users accidentally called `subscriber.error` twice. Likewise, would we report an error
+     * for two calls of `complete`? This is really something a build-time tool like a linter should
+     * capture. Not a run time error reporting event.
+     */
```

Comment:
> It's part of the contract of observables

AFAICT, [the contract](http://reactivex.io/documentation/contract.html) stipulates how the communication between the observable and the observer is to occur. It says nothing about how errors ought to be treated when then communication channel between the observable and the observer has been closed.

IMO, this behaviour really ought to be configurable. Intentionally swallowing errors when a reporting channel is known to be closed is less than ideal. If a config option were there, I'd definitely be enabling it.

> This is really something a build-time tool like a linter should capture. Not a run time error reporting event.

FWIW, for all but the most trivial scenarios, this is beyond the capabilities of static analysis.

Reply:
I have some ideas about how this could be addressed. I'll create an issue or a PR after I've given it more thought.

Reply:
The only important thing is that we don't break anyone.

Reply:
Sure. The change that I'm contemplating is independent of the `onUnhandledError` added in this PR, so this is good to merge, IMO.